horovod/common/operations.cc

```diff
@@ -1305,29 +1311,52 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   auto mpi_threads_disable = std::getenv("HOROVOD_MPI_THREADS_DISABLE");
   int required = MPI_THREAD_MULTIPLE;
   if (mpi_threads_disable != nullptr &&
-      std::strtol(mpi_threads_disable, nullptr, 10) > 0) {
+    std::strtol(mpi_threads_disable, nullptr, 10) > 0) {
     required = MPI_THREAD_FUNNELED;
   }
   int provided;
-  MPI_Init_thread(NULL, NULL, required, &provided);
+  int is_mpi_initialized = 0;
+  MPI_Initialized(&is_mpi_initialized);
+  if (is_mpi_initialized) {
+    MPI_Query_thread(&provided);
+  } else {
```

Comment:
Let's add a warning - if we did not initialize mpi, and multi threading is not supported, this are very likely going to fail with segfault.